backend/main.py

```diff
@@ -113,6 +113,36 @@ def new_game():
   return jsonify(firebase.put('/games', game, put_data))
 
 
+@app.route('/joinGame', methods=['POST'])
+def join_game():
+  request_data = request.get_json()
+  game = request_data['gameId']
+  player = request_data['playerId']
+  name = request_data.get('name', '')
+  need_gun = request_data.get('needGun', False)
+  profile_image_url = request_data.get('profileImageUrl', '')
+  start_as_zombie = request_data.get('startAsZombie', False)
+  volunteer = request_data.get('volunteer', False)
+  be_secret_zombie = request_data.get('beSecretZombie', False)
+  user_id = request.headers['userId']
```

Comment:
We put the userId into the header because it's included in literally every request, and in most of them, it's not part of the semantics, but instead part of the security. For example, when posting a message to a chat room, the userId is irrelevant except for checking security. The pattern would have been to put anything semantically relevant into the body, and anything else into the header (and if a userId was semantically important, maybe we'd put it in both, I dont know).

That *seems* to be roughly the pattern that HTTP follows, but thats just my best guess. Seemed clean to me, but "clean" is in the eye of the beholder, and I don't think I have any solid reasons to go either way. I think this particular decision has more of an effect on server code than client, so your call =) I'll update client to match.